muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/reducers/main-function/calendar.ts

```diff
@@ -0,0 +1,144 @@
+import { ActionType } from "~/actions";
+
+/**
+ *  Participant
+ */
+export type Participant = {
+  userEntityId: number;
+  name: string;
+  attendance: "UNCONFIRMED" | "YES" | "NO" | "MAYBE";
+};
+
+export type eventDisplay =
+  | "auto"
+  | "block"
+  | "list-item"
+  | "background"
+  | "inverse-background"
+  | "none";
+/**
+ *  CalendarEvent
+ */
+export interface CalendarEvent {
+  id?: number;
+  title?: string;
+  start: string;
+  end?: string;
+  overlap?: boolean; // Can the event overlap with others
+  editable?: boolean; // Override for the event
+  classNames?: string[];
+  description?: string;
+  display?: eventDisplay; // How the event is displayed in the calendar
+  backgroundColor?: string;
+  resourceId?: string; // Resource
+  participants?: Participant[];
+}
```

Comment:
With props documentation use /* */ style so code intelligence can catch that also.